runtime/queries/metricsview_aggregation.go

```diff
@@ -246,16 +407,30 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 		limitClause = fmt.Sprintf("LIMIT %d", *q.Limit)
 	}
 
-	sql := fmt.Sprintf("SELECT %s FROM %s %s %s %s %s %s OFFSET %d",
-		strings.Join(selectCols, ", "),
-		safeName(mv.Table),
-		strings.Join(unnestClauses, ""),
-		whereClause,
-		groupClause,
-		orderClause,
-		limitClause,
-		q.Offset,
-	)
+	var sql string
+	if q.PivotOn != nil {
+		// select m1, m2, d1, d2 from t, lateral unnest(t.d1) tbl(unnested_d1_) where d1 = 'a' group by d1, d2
+		sql = fmt.Sprintf("SELECT %[1]s FROM %[2]s %[3]s %[4]s %[5]s %[6]s %[7]s",
+			strings.Join(selectCols, ", "),  // 1
+			safeName(mv.Table),              // 2
+			strings.Join(unnestClauses, ""), // 3
+			whereClause,                     // 4
+			groupClause,                     // 5
+			orderClause,                     // 6
+			limitClause,                     // 7
+		)
+	} else {
+		sql = fmt.Sprintf("SELECT %s FROM %s %s %s %s %s %s OFFSET %d",
+			strings.Join(selectCols, ", "),
+			safeName(mv.Table),
+			strings.Join(unnestClauses, ""),
+			whereClause,
+			groupClause,
+			orderClause,
+			limitClause,
+			q.Offset,
+		)
+	}
```

Comment:
They seem equivalent apart from the offset clause? I would suggest we return an error if passing `q.Offset` together with `q.PivotOn`, since that's not easy to support right now